support-frontend/assets/components/page/offerStrapline.tsx

```diff
@@ -0,0 +1,100 @@
+import { css } from '@emotion/react';
+import { from } from '@guardian/source-foundations';
+import type { ReactElement, ReactNode } from 'react';
+import { offerStraplineBlue } from 'stylesheets/emotion/colours';
+
+const offerStraplineStyles = css`
+	font-family: GuardianTextSans, 'Helvetica Neue', Helvetica, Arial,
+		'Lucida Grande', sans-serif;
+	font-size: 17px;
+	font-weight: bold;
+	padding: 4px 10px 8px;
+	max-width: 100%;
+	width: fit-content;
+	margin-bottom: 0;
+	background-color: ${offerStraplineBlue};
+
+	${from.phablet} {
+		padding: 4px 20px 8px;
+	}
+	${from.tablet} {
+		font-size: 20px;
+		padding: 4px 20px 12px;
+		max-width: 50%;
+	}
+	${from.desktop} {
+		font-size: 24px;
+		max-width: 50%;
+	}
+`;
+
+// Requirement: strapline acts differently (becomes full-width) at smaller device widths if the copy is longer than 32 chars
+const offerStraplineLongCopyStyles = css`
```

Comment:
this is very similar to `offerStraplineStyles`. How about a function to reduce duplication?
```
const offerStraplineStyles = (isLong: boolean) => css`
  ${textSans.medium({
    fontWeight: 'bold',
  })}
	padding: 4px 10px 8px;
	margin-bottom: 0;
	background-color: ${offerStraplineBlue};
	${isLong
		? 'width: 100%;'
		: `
	  width: fit-content;
	  max-width: 100%;
  `}

	${from.phablet} {
	  padding: 4px 20px 8px;
		${isLong
			? `
	    width: fit-content;
	    max-width: 100%;
	  `
			: ''}
	}
	${from.tablet} {
		font-size: 20px;
		padding: 4px 20px 12px;
		max-width: 50%;
	}
	${from.desktop} {
		font-size: 24px;
		max-width: 50%;
	}
`;
```

Reply:
Good suggestion. Implemented.